common/cpp/crypto/crypto_utils.cpp

```diff
@@ -52,28 +51,60 @@ namespace Error = tcf::error;
 
 /**
  * Generate a cryptographically strong random bit string.
+ * Implemented with mbedtls_ctr_drbg_random().
  * Throws: RuntimeError.
  */
 ByteArray pcrypto::RandomBitString(size_t length) {
     char err[constants::ERR_BUF_LEN];
     ByteArray buf(length);
-    int res = 0;
+    int res;
 
     if (length < 1) {
         std::string msg("Crypto Error (RandomBitString): "
             "length argument must be at least 1");
         throw Error::ValueError(msg);
     }
 
-    res = RAND_bytes(buf.data(), length);
+#ifdef USE_MBEDTLS
```

Comment:
If we are planning to add more mbedtls crypto functions in this file then I would suggest to create a different file for easier maintenance.  Unfortunately Avalon C++ Crypto implementation is not really object oriented else we should could use some design pattern like factory pattern to create OpenSSL or MbedTLS  crypto utils class

Reply:
OK, I will have separate files for OpenSSL/Mbed TLS.
I will first file another PR to separate the OpenSSL-dependent and independent parts of the .cpp implementation.

Reply:
@manojgop I updated the PR. I separated the Mbed TLS code from OpenSSL code.
More Mbed TLS code is coming--this is the first step.
Please re-review.